core/tee/tee_svc.c

```diff
@@ -443,16 +431,13 @@ static TEE_Result tee_svc_copy_param(struct tee_ta_session *sess,
  */
 static TEE_Result tee_svc_update_out_param(
 		struct tee_ta_session *sess,
-		struct tee_ta_session *called_sess,
 		struct tee_ta_param *param,
 		tee_paddr_t tmp_buf_pa[TEE_NUM_PARAMS],
 		struct abi_user32_param *usr_param)
 {
 	size_t n;
 	TEE_Param callee_params[TEE_NUM_PARAMS];
-	bool have_private_mem_map = (called_sess == NULL) ||
-		(called_sess->ctx->static_ta != NULL) ||
-		((called_sess->ctx->flags & TA_FLAG_USER_MODE) != 0);
+	bool have_private_mem_map = true;
```

Comment:
Agree, but we don't have kernel TAs any longer. This variable should be removed either in this patch or in separate kernel TA cleanup patch whatever you prefer.